configs/components/rubygem-nokogiri.rb

```diff
@@ -5,14 +5,18 @@
     # including its dependencies (mini_portile2), installing it on the target, and
     # installing the nokogiri gem using our gem command. The added files from
     # :gem_home were then tar'ed up and packed into this container tarball.
-    pkg.url "http://buildsources.delivery.puppetlabs.net/nokogiri-precompiled-huaweios-#{pkg.get_version}.tar.gz"
-    pkg.md5sum "8a783bdda33cf5f10afe84cf6fd8f28c"
+    pkg.url "http://buildsources.delivery.puppetlabs.net/nokogiri-precompiled-huaweios-#{pkg.get_version}-for-ruby-#{settings[:ruby_version]}.tar.gz"
+    if settings[:ruby_version] == "2.1.9"
```

Comment:
Yeah, we had many discussions about how to clean up components in RE, and I think the conclusions we came to were good. We wanted to revisit the cleanliness of components and better ways to remove some conditionals, but not until we updated all the configs to use the styleguide. That way when all the components were using the same styleguide we could identify patterns that we could eliminate.